lib/sycamore/sycamore/connectors/qdrant/qdrant_writer.py

```diff
@@ -0,0 +1,132 @@
+
+        self._client.upload_points(collection_name=target_params.collection_params["collection_name"], points=points)
+
+    def create_target_idempotent(self, target_params: "BaseDBWriter.TargetParams"):
+        assert isinstance(target_params, QdrantWriterTargetParams)
+        params = target_params.collection_params
+        if not self._client.collection_exists(params["collection_name"]):
+            self._client.create_collection(**params)
+
+    def get_existing_target_params(self, target_params: "BaseDBWriter.TargetParams") -> QdrantWriterTargetParams:
+        assert isinstance(target_params, QdrantWriterTargetParams)
+
+        collection_info = self._client.get_collection(target_params.collection_params["collection_name"])
+        vectors_config = collection_info.config.params.vectors
+        assert vectors_config is not None
+        return QdrantWriterTargetParams(
+            collection_params={
+                "collection_name": target_params.collection_params["collection_name"],
+                "vectors_config": (
+                    vectors_config
+                    if isinstance(vectors_config, dict)
+                    else vectors_config.model_dump(exclude_defaults=True)
+                ),
+                "sparse_vectors_config": collection_info.config.params.sparse_vectors,
+                "shard_number": collection_info.config.params.shard_number,
+                "sharding_method": collection_info.config.params.sharding_method,
+                "replication_factor": collection_info.config.params.replication_factor,
+                "write_consistency_factor": collection_info.config.params.write_consistency_factor,
+                "on_disk_payload": collection_info.config.params.on_disk_payload,
+                "hnsw_config": collection_info.config.hnsw_config,
+                "optimizers_config": collection_info.config.optimizer_config,
+                "wal_config": collection_info.config.wal_config,
+                "quantization_config": collection_info.config.quantization_config,
+            },
+        )
+
+
+@dataclass
+class QdrantWriterRecord(BaseDBWriter.Record):
+    id: str
+    vector: Union[list[float], dict[str, list[float]]]
+    payload: dict[str, Any]
+
+    @classmethod
+    def from_doc(cls, document: Document, target_params: "BaseDBWriter.TargetParams") -> "QdrantWriterRecord":
+        assert isinstance(target_params, QdrantWriterTargetParams)
+        assert document.doc_id is not None, f"Document found with null id: {document}"
+        vector: Union[list, dict] = document.embedding or {}
+        payload = {
+            "type": document.type,
+            "text_representation": document.text_representation,
+            "bbox": document.bbox.to_dict() if document.bbox else None,
+            "shingles": [str(s) for s in document.shingles] if document.shingles else None,
```

Comment:
from reading docs it looks like qdrant is happy to store lists of ints - so you prob don't need to stringify the shingles. [This](https://www.aryn.ai/post/near-duplicate-detection-in-sycamore-what-is-it-good-for) is the use-case for them btw... all that really matters is that you can to equality checks at query-time